### PR TITLE
docker-compose subgenerator: uaa support

### PR DESCRIPTION
**Pull request by pascalgrimaud**
related to https://github.com/jhipster/generator-jhipster/issues/4395

### CODE REVIEW

File: generators/docker-compose/prompts.js [hunk 1]
```diff
@@ -287,7 +287,8 @@ function getAppFolders(input, applicationType) {
                     if ((fileData['generator-jhipster'].baseName !== undefined)
                         && ((applicationType === undefined)
                             || (applicationType === fileData['generator-jhipster'].applicationType)
-                            || ((applicationType === 'microservice') && ('gateway' === fileData['generator-jhipster'].applicationType)))) {
+                            || ((applicationType === 'microservice') && ('gateway' === fileData['generator-jhipster'].applicationType))
```
Comment (xetys) [inline generators/docker-compose/prompts.js]: shouldn't it be `applicationType === 'uaa'`?

Comment (pascalgrimaud) [inline generators/docker-compose/prompts.js]: Oh I see, it's a little bit confusing. This `applicationType` is a parameter in `function getAppFolders(input, applicationType) {`, and his value can be: monolith or microservice, accordingly to the first question: monolith or microservice

**Comment by pascalgrimaud** [comment]
When using 30sec or 60sec, when the gateway start, sometimes, it failed because it seems like the uaa is not completely registered. With 80sec, it always works for me.
I don't know if it's my computer or the mecanism in this architecture, but I prefer letting time and be sure the uaa is up, so that everything works fine. Every users can also reduct this time

File: generators/docker-compose/prompts.js (hunk 1 above)
Comment (xetys) [inline generators/docker-compose/prompts.js]: it can be 'uaa', too. This actually is why the generator seems not to generate the yaml

Comment (pascalgrimaud) [inline generators/docker-compose/prompts.js]: so for me, it's correct, but don't hesitate to test it!

**Comment by PierreBesson** [comment]
Although those sleep delays on initialization are hackish we don't have another way because of the limitations of docker-compose.

However for Kubernetes there is a feature currently in beta called init containers that might solve our problem :
http://kubernetes.io/docs/user-guide/production-pods/#handling-initialization

File: generators/docker-compose/prompts.js (hunk 1 above)
Comment (pascalgrimaud) [inline generators/docker-compose/prompts.js]: Using console.log to display variables:

```
╭─pgrimaud@pgrxps  ~/test/03-eureka-uaa/00-compose  
╰─$ jhcompose
Welcome to the JHipster Docker Compose Sub-Generator 
Files will be generated in folder: /home/pgrimaud/test/03-eureka-uaa/00-compose

Found .yo-rc.json config file...
? Which *type* of application would you like to deploy? Microservice application
? Enter the root directory where your gateway(s) and microservices are located (../) applicationType=microservice and fileData['generator-jhipster'].applicationType=uaa
applicationType=microservice and fileData['generator-jhipster'].applicationType=gateway
? Enter the root directory where your gateway(s) and microservices are located ../
applicationType=microservice and fileData['generator-jhipster'].applicationType=uaa
applicationType=microservice and fileData['generator-jhipster'].applicationType=gateway
2 applications found at /home/pgrimaud/test/03-eureka-uaa/

? Which applications do you want to include in your configuration? (Press <space> to select, <a> to toggle all, <i> to inverse selection)
❯◉ 01-uaa
 ◉ 02-jhgateuaa
```

Comment (pascalgrimaud) [inline generators/docker-compose/prompts.js]: @xetys : I rename the variable, to avoid confusing, is it ok for u?

**Pull request merged**